test/integration/controllers/leaderworkerset_test.go

```diff
@@ -1971,6 +1971,41 @@ var _ = ginkgo.Describe("LeaderWorkerSet controller", func() {
 				},
 			},
 		}),
+		ginkgo.Entry("resize shoud update the size of the replicas", &testCase{
+			makeLeaderWorkerSet: func(nsName string) *wrappers.LeaderWorkerSetWrapper {
+				return wrappers.BuildLeaderWorkerSet(nsName).Replica(2).Size(2).ResizePolicy(leaderworkerset.ResizePolicyRecreate)
+			},
+			updates: []*update{
+				{
+					// Set lws to available condition.
+					lwsUpdateFn: func(lws *leaderworkerset.LeaderWorkerSet) {
+						testing.SetPodGroupsToReady(ctx, k8sClient, lws, 2)
+					},
+					checkLWSState: func(lws *leaderworkerset.LeaderWorkerSet) {
+						testing.ExpectLeaderWorkerSetAvailable(ctx, k8sClient, lws, "All replicas are ready")
+						testing.ExpectStatefulsetPartitionEqualTo(ctx, k8sClient, lws, 0)
+						testing.ExpectValidLeaderStatefulSet(ctx, k8sClient, lws, 2)
+						testing.ExpectValidWorkerStatefulSets(ctx, lws, k8sClient, true)
+						testing.ExpectLeaderWorkerSetStatusReplicas(ctx, k8sClient, lws, 2, 2)
+					},
+				},
+				{
+					// Update size to 3.
+					lwsUpdateFn: func(lws *leaderworkerset.LeaderWorkerSet) {
+						testing.UpdateSize(ctx, k8sClient, lws, 3)
+						testing.SetPodGroupsToReady(ctx, k8sClient, lws, 2)
+					},
+					checkLWSState: func(lws *leaderworkerset.LeaderWorkerSet) {
```

Comment:
Nit: let's verify that before the size is 2, after update the size increased to 3, I think we missed that part.

Reply:
I would say that `ExpectValidWorkersStatefulSets` being called before and after the resize takes care of that ? [Here](https://github.com/kubernetes-sigs/lws/blob/8d65262f630745031013318c8861d22fb892e738/test/testutils/validators.go#L286-L288) and [here](https://github.com/kubernetes-sigs/lws/blob/8d65262f630745031013318c8861d22fb892e738/test/testutils/validators.go#L286-L288) for the size annotation.

Reply:
Although not that straightforward, but fair enough, since we assume that the update is succeeded.